processing/src/main/java/io/druid/query/dimension/DimensionSpec.java

```diff
@@ -22,23 +22,30 @@
 import com.fasterxml.jackson.annotation.JsonSubTypes;
 import com.fasterxml.jackson.annotation.JsonTypeInfo;
 import io.druid.query.extraction.ExtractionFn;
+import io.druid.segment.DimensionSelector;
 
 /**
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type", defaultImpl = LegacyDimensionSpec.class)
 @JsonSubTypes(value = {
     @JsonSubTypes.Type(name = "default", value = DefaultDimensionSpec.class),
-    @JsonSubTypes.Type(name = "extraction", value = ExtractionDimensionSpec.class)
+    @JsonSubTypes.Type(name = "extraction", value = ExtractionDimensionSpec.class),
+    @JsonSubTypes.Type(name = "regexFiltered", value = RegexFilteredDimensionSpec.class),
```

Comment:
I'm not sure how to improve things without a much larger reworking. My concern is that anyone using Druid for the first time will find it very confusing to know what specific APIs to use for every specific situation. My suggestion right now is to expand the section on Multi-value dimensions and include the fact that these filters should be used to optimize performance.